### PR TITLE
Guardians (Holo and holoclown) have syndicate faction ai

### PR DESCRIPTION
**Pull request by QuietlyWhisper**
## About the PR

Removed the player requirement, the ghost takeover still works.
Players can summon their guardians without requiring someone to be playing it (This is useful for lowpop and low death count)
Players will still see the notification when a soul takes over the holo
When the holo is ghosted, it returns to AI control
Non syndicate cannot use uplink guardians; they will attack anyone that isn't a syndicate (Crew members shouldnt be using contraband that is likely loyal to the syndicate)

## Why / Balance

Guardians can be an entire waste of telecrystals when they rely on ghosts that may or may not exist. This removes the gambling aspect. 
Should encourage more guardian usage by syndicate agents when they are guaranteed to get usage out of it.
Allows usage on lowpop where ghosts and ghosts willing to play the role are scarce.

## Technical details

Added the syndicate faction and AI components to the uplink guardians.
Removed the check for player control, so the guardian may always be summoned.

## Media

(discord links to the ss14 server, videos are too large for github)
https://discord.com/channels/310555209753690112/675078881425752124/1148247722260963349
https://discord.com/channels/310555209753690112/675078881425752124/1148249548393816064
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Whisper
- add: Guardians (holo, holoclown) have AI control before a ghost takes over the role, and after a player ghosts from it.
- tweak: Guardians can be summoned without a player in the role.

### CODE REVIEW

**Comment by deltanedas** [comment]
would be cool to see if the salvager ai compound for taking weapons could be used for holoclown to make its ai more useful

**Comment by QuietlyWhisper** [comment]
> would be cool to see if the salvager ai compound for taking weapons could be used for holoclown to make its ai more useful

it's included. Though clowns aren't very good with guns :)

**Comment by deltanedas** [comment]
:trollface:

**Comment by JoeHammad1844** [comment]
I was banned from the discord, please link youtube videos

File: Content.Server/Guardian/GuardianSystem.cs [hunk 1]
```diff
@@ -301,10 +301,10 @@ private void OnGuardianMove(EntityUid uid, GuardianComponent component, ref Move
                 RetractGuardian(hostUid, hostComponent, guardianUid, guardianComponent);
         }
 
-        private bool CanRelease(EntityUid guardian)
+        /*private bool CanRelease(EntityUid guardian)
```
Comment (EmoGarbage404) [inline Content.Server/Guardian/GuardianSystem.cs:304]: remove commented out code

File: Content.Server/Guardian/GuardianSystem.cs [hunk 2]
```diff
@@ -314,11 +314,11 @@ private void ReleaseGuardian(EntityUid host, GuardianHostComponent hostComponent
                 return;
             }
 
-            if (!CanRelease(guardian))
+            /*if (!CanRelease(guardian))
```
Comment (EmoGarbage404) [inline Content.Server/Guardian/GuardianSystem.cs:317]: here too

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by QuietlyWhisper** [comment]
should I be worried robust is in the changed files tab for some reason?

**Comment by deltanedas** [comment]
yes
`git checkout upstream/master RobustToolbox` and commit it (either that or reset i forget which)

**Review by DrSmugleaf — DISMISSED** [review]

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**